retrievalmarket/impl/blockunsealing/blockunsealing.go

```diff
@@ -92,3 +87,22 @@ func (lu *loaderWithUnsealing) attemptUnseal() error {
 
 	return nil
 }
+
+func (lu *loaderWithUnsealing) firstSuccessfulUnseal(cidInfo piecestore.CIDInfo) (io.ReadCloser, error) {
```

Comment:
so the logic changed a bit here. theoretically since we're basing our queries on payload CIDs, I guess it's possible that a query could span multiple pieces now (if a piece was a partial but not complete DAG -- something we intend to support I believe). 

The point of this function is: for a given Cid, find the first piece that it's contained in that we can successfully unseal.

since we unseal and read the whole piece, in theory this function should not get called unless the query we make somehow crosses a piece boundary, which again, I think is possible.

If unsealing fails completely the first time, it's probably alright cause the loader calling code will immediately error and stop the retrieval deal.